### DISCUSSION TITLE
Accessing other fields from a row selected in an option picker

### DISCUSSION THREAD

**Discussion by Dave-Perkins-github** [Budibase/budibase#7520] [Help] [answered]
Is there a way to access other fields from a row selected
in an option picker?

I have a driver view that includes a Driver ID column
and a Truck Type column.

I have an option picker that the user uses to select the Driver ID
of the driver they want.  I want to use the Truck Type column from
the row they selected to populate a text field on my form.

Is this possible?

Happy to post the export if you want it.

**Comment by andz-bb** [comment]
hey @Dave-Perkins-github, could you send the app export to support@budibase.com please? I think this should be possible but just would like to see how the data is structured in your case. Thanks

**Comment by Dave-Perkins-github** [comment]
Done.  And it came from this email address.

**Reply by andz-bb** [reply]
Thanks @Dave-Perkins-github, I will have a look. Is it the Create and Update Hopper screens where you would like to have the Truck data populate based on the Driver selection?

**Comment by Dave-Perkins-github** [comment]
Thanks for your time.  Those are exactly where I'm looking for a way to do
this.  I'm guessing there has to be
a way to save the row ID or something when the driver is chosen, so we
don't have to look up the row again
to get the truck type.

**Comment by andz-bb** [comment, marked as answer]
Hi @Dave-Perkins-github 

I was able to get this working using an Update Field Value action on the Driver Picker.

<img width="1955" alt="Screenshot 2022-09-05 at 10 09 28" src="https://user-images.githubusercontent.com/110921612/188412921-e80d427e-c06a-484c-bd5a-374f7a6c23d0.png">

I used this JS function to return the value we're looking for based on the driver selected.

```
const selectedDriver = $("drivers_v.Rows").find(x => x["Driver ID"] === $("Field Value"))
return selectedDriver.truck_type_abbr;
``` 

Hopefully this helps! If you need a hand with anything else or want me to explain what I did in more detail let me know!

**Reply by Dave-Perkins-github** [reply]
Thanks andz-bb.  I've set it up this way, and it's throwing a JS error.  Any tips on how to debug it?
![jlst_js_error](https://user-images.githubusercontent.com/77355307/188671525-891d2e58-3ad1-405b-a80d-1531c20e6994.png)

**Reply by andz-bb** [reply]
Hey @Dave-Perkins-github

can you paste in the JS you're using for that please? It might be that the bindings for the values you want are different on the _Update Hopper_ page vs the _New Hopper_ page. You would want to make sure that the Data Provider for Drivers on the _Hopper Update_ page is called `drivers_v` the way it is on the _New Hopper_ page if you want the binding `$("drivers_v.Rows")` to work, otherwise you will need to update drivers_v to the name of the relevant Data Provider in that case.

**Reply by Dave-Perkins-github** [reply]
Nailed it!!!!  On the Update Hopper page, the view is named driver_v, not drivers_v in the JS you supplied.  I just changed it to $("driver_v.Rows") and it's working perfectly!  Thank you!!!

**Reply by Dave-Perkins-github** [reply]
And with modified JS code, it's now working on both pages.

**Comment by Dave-Perkins-github** [comment]


**Reply by andz-bb** [reply]
Hi @Dave-Perkins-github, sorry to hear you're still having problems with this. Could you send me your most recent app export? I'll try and figure out what's going wrong. Thanks

**Comment by Dave-Perkins-github** [comment]
I did not post that update.  I was not even awake 5 hours ago.  This was
solved (as I marked it answered) back in September.
I suspect my github account has been compromised.
I just changed my password and removed the duplicate comment.
